aladdin/lib/aws/certificate.py

```diff
@@ -71,38 +71,32 @@ def new_certificate_arn(boto_session, dns_name):
     return ""
 
 
-def _validate_certificate_with_retry(boto_session, dns_name, arn, max_retries=20):
+def _validate_certificate_with_retry(boto_session, arn, max_retries=20):
     """Validate new certificate using DNS validation"""
     log = logging.getLogger(__name__)
-
-    # Create the hosted zone for later dns validation
-    hostedzone_name = ".".join(dns_name.split(".")[1:])
-    hostedzone = get_hostedzone(boto_session, hostedzone_name) or create_hostedzone(
-        boto_session, hostedzone_name
-    )
-
     acm = boto_session.client("acm")
 
+    cname_record = None
     retry_count = 0
-    while retry_count < max_retries:
+    while not cname_record and retry_count < max_retries:
         try:
             cname_info = acm.describe_certificate(CertificateArn=arn)["Certificate"][
                 "DomainValidationOptions"
             ][0]["ResourceRecord"]
             cname_record = {cname_info["Name"][:-1]: cname_info["Value"]}
         except (KeyError, IndexError):
             retry_count += 1
-        else:
-            break
-    if retry_count == max_retries:
+
+    if cname_record:
+        # Create the hosted zone and add the cname used for dns validation
+        fill_hostedzone(boto_session, cname_record)
```

Comment:
this is the core change, previously we used `fill_dns_dict` which correctly setup the CNAME for DNS validation, but it did not correctly setup delegation of the newly created hosted zone. `fill_hostedzone` does all of this